js_src/rdm.js

```diff
@@ -5,146 +5,969 @@ angular.module('rdmApp', [])
+    // If as_hex isn't defined we try to be clever about what we accept.
+    // ##h and 0x## are obviously hex values, ## is ambiguous so we use
+    // heuristics to figure it out.
+    var error = '';
+    var binary_data = [];
+
+    var tokens = [];
+    for (var i = 0; i < lines.length; ++i) {
+     var line = lines[i].replace(/^(.*?)\s*\/\/.*$/, '$1');
+     line = line.replace(/[,:\-]/g, ' ');
+     line = line.replace(/\s{2,}/g, ' ').trim();
+     if (line) {
+      tokens = tokens.concat(line.split(' '));
+     }
+    }
+
+    if (as_hex === undefined) {
+     as_hex = guessDataFormat(tokens);
+    }
+
+    $log.info(tokens);
+    for (var j = 0; j < tokens.length; j++) {
+     var token = tokens[j];
+     var hex_suffix_match = token.match(/^([\da-fA-F]{1,2})h$/);
+     if (hex_suffix_match) {
+      binary_data.push(parseInt(hex_suffix_match[0], 16));
+      continue;
+     }
+
+     var hex_prefix_match = token.match(/^0x([\da-fA-F]{1,2})$/);
+     if (hex_prefix_match) {
+      binary_data.push(parseInt(hex_prefix_match[0], 16));
+      continue;
+     }
+
+     if (as_hex) {
+      var hex_match = token.match(/^([\da-fA-F]{1,2})$/);
+      if (hex_match) {
+       binary_data.push(parseInt(hex_match[0], 16));
+       continue;
+      } else {
+       error = 'Invalid byte: ' + token;
+       return [error, binary_data];
+      }
+     } else {
+      var decimal_match = token.match(/^(\d{1,3})$/);
+      if (decimal_match) {
+       if (decimal_match[0] <= 255) {
+        binary_data.push(parseInt(decimal_match[0], 10));
+        continue;
+       } else {
+        error = 'Invalid byte: ' + token;
+        return [error, binary_data];
+       }
+      }
+     }
+
+     error = 'Invalid binary data: ' + token;
+    }
+    return [error, binary_data];
+   };
+
+   this.textToBytes = function(text, as_hex) {
+     var lines = text.split('\n');
+     return parseLines(lines, as_hex);
+    };
+
+   this.uidToBytes = function(text) {
+    var clean_uid = text.replace(/[\.:\-\s]/g, '');
+    if (!clean_uid.match(/^[0-9a-fA-F]{12}$/)) {
+     return ['Contains non hex characters', []];
+    }
+
+    if (clean_uid.length !== RDM.UID_SIZE * 2) {
+     return ['UID should be ' + RDM.UID_SIZE + ' bytes', []];
+    }
+
+    var uid_bytes = [];
+    for (var i = 0; i < clean_uid.length / 2; i++) {
+     var octet = clean_uid.slice(i * 2, (i + 1) * 2);
+     var value = parseInt(octet, 16);
+     if (isNaN(value)) {
+      return ['Invalid UID: bad value' + octet, []];
+     }
+     uid_bytes.push(value);
+    }
+
+    return ['', uid_bytes];
+   };
+  }])
+
+ .controller('UIDController',
+             ['$scope', '$log', 'checksumService', 'formatService',
+              'parserService',
+              function($scope, $log, checksumService, formatService,
+                       parserService) {
   'use strict';
-  convertor.invalid_input_message = 'Invalid UID, please enter a UID in the ' +
+  var OUTPUT_FORMATS = {
```

Comment:
It should, but it's only used in this controller so I didn't want to put it in with the RDM constant.

I think what I'm going to do after this change is in is split these up into separate modules.